src/components/accordion/Accordion.tsx

```diff
@@ -0,0 +1,83 @@
+import type { ReactNode} from "react";
+import { useState } from "react"
+import AccordionItem from "./AccordionItem"
+
+export type AccordionData = {
+  title: string
+  content: ReactNode
+}
+
+export type ShowMoreLabelProps = {
+  toggle: () => void
+  items: Array<any>
+  show: boolean
+}
+
+function Accordion({
+  items,
+  itemsCutoff,
+  multipleOpen,
+  openByDefault,
+  titleLabels,
+  currentState,
+  actionClickLabels,
+}: {
+  items: Array<AccordionData>
+  itemsCutoff: number
+  multipleOpen: boolean
+  openByDefault: boolean
+  titleLabels?: boolean
+  currentState?: Array<boolean>
+  actionClickLabels?: (id: number) => void
+}) {
+  const [currentIdx, setCurrentIdx] = useState(new Array<number>())
+  const [showFullList, setShowFullList] = useState(false)
+  const btnOnClick = (idx: number) => {
+    multipleOpen
+      ? setCurrentIdx((currentValue) =>
+          currentValue.includes(idx) ? currentValue.filter((item) => item !== idx) : [...currentValue, idx]
+        )
+      : setCurrentIdx((currentValue) => (currentValue.includes(idx) ? [] : [idx]))
+  }
+  const cutItems = showFullList ? items : items.slice(0, itemsCutoff)
+  if (openByDefault && !multipleOpen) {
+    setCurrentIdx([0]);
+  }
+  return (
+    <ul className="block m-0 pl-4 overflow-x-hidden">
+      {cutItems.map((item, idx) => (
+        <>
+          <AccordionItem
+            key={item.title + idx + "--accordion"}
+            data={item}
+            isOpen={
+              currentState && currentState[idx]
+                ? currentState[idx]
+                : openByDefault && multipleOpen
+                ? !currentIdx.includes(idx)
+                : currentIdx.includes(idx)
+            }
+            titleLabels={titleLabels}
+            btnOnClick={() => {
+              btnOnClick(idx)
+              if (actionClickLabels) actionClickLabels(idx);
+            }}
+          />
+          <br />
```

Comment:
It seems there are some spacing issues in the commit view.
![image](https://github.com/git-truck/git-truck/assets/1959615/89836e3f-d7de-41e6-a878-0c39fac5ea21)

This could be solved with `display: flex` and using the `gap` property for specifying spacing between the entries.

Reply:
I put here just a line break `<br />` before. Now I removed it. Do you think it is better now? If not, what size should the gap be?